core/peerset/src/slots.rs

```diff
@@ -182,13 +183,49 @@ impl Slots {
 	pub fn is_reserved(&self, peer_id: &PeerId) -> bool {
 		self.reserved.contains_key(peer_id)
 	}
+
+	/// Produces a JSON object containing the state of slots, for debugging purposes.
+	pub fn debug_info(&self) -> serde_json::Value {
+		json!({
+			"max_slots": self.max_slots,
+			"reserved": self.reserved.keys().map(|peer_id| peer_id.to_base58()).collect::<Vec<_>>(),
+			"common": self.common.keys().map(|peer_id| peer_id.to_base58()).collect::<Vec<_>>()
+		})
+	}
 }
 
 #[cfg(test)]
 mod tests {
 	use libp2p::PeerId;
+	use serde_json::to_string_pretty;
 	use super::{Slots, SlotType};
 
+	#[test]
+	fn test_slots_debug_info() {
+		let reserved_peer = PeerId::random();
+		let reserved_peer2 = PeerId::random();
+		let common_peer = PeerId::random();
+		let mut slots = Slots::new(10);
+
+		slots.add_peer(reserved_peer.clone(), SlotType::Reserved);
+		slots.add_peer(reserved_peer2.clone(), SlotType::Reserved);
+		slots.add_peer(common_peer.clone(), SlotType::Common);
+
+		let expected = format!(r#"{{
```

Comment:
Instead of building this json manually, we could look into the `serde_json::Value`?

Or have the structure defined in the test and use `serde_json` to deserailize it?